controllers/druid/druid_client.go

```diff
@@ -0,0 +1,76 @@
+package druid
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+)
+
+type DruidClient struct {
+	BaseUrl  string
+	UserName string
+	PassWord string
+	client   *http.Client
+}
+
+func InitClient(BaseUrl string, UserName string, PassWord string) DruidClient {
+	d := DruidClient{BaseUrl, UserName, PassWord, &http.Client{}}
+	return d
+}
+
+func (d DruidClient) GetCoordinatorConfig() (map[string]interface{}, error) {
+	req, err := http.NewRequest("GET", d.BaseUrl+"/druid/coordinator/v1/config", nil)
```

Comment:
Add retries to this request with a max retry set.

Reply:
The changes suggested for this method applies to all the other methods which are making HTTP requests.

Reply:
Add a helper function to prepare the URL instead of ```d.BaseUrl+"/druid/coordinator/v1/config",```

Also, add routes like `/druid/coordinator/v1/config` to a constant from where the URL can be fetched and built. This way there would be less margin for error